domains/client/relayer/src/lib.rs

```diff
@@ -394,38 +363,36 @@ where
     }
 
     /// Constructs the proof for the given key using the domain backend.
-    fn construct_domain_storage_proof_for_key_at<CNumber, CHash>(
-        consensus_chain_block_info: BlockInfo<CNumber, CHash>,
+    fn construct_domain_storage_proof_for_key_at<CHash>(
+        consensus_chain_block_hash: CHash,
         domain_client: &Arc<Client>,
         block_hash: Block::Hash,
         key: &[u8],
-        consensus_chain_state_root: CHash,
+        _consensus_chain_state_root: CHash,
         domain_proof: StorageProof,
     ) -> Result<ProofOf<Block>, Error>
     where
-        CNumber: Into<NumberFor<Block>>,
         CHash: Into<Block::Hash>,
     {
         domain_client
             .header(block_hash)?
             .map(|header| (*header.number(), header.hash()))
-            .and_then(|(number, hash)| {
+            .and_then(|(_number, _hash)| {
                 let proof = domain_client
                     .read_proof(block_hash, &mut [key].into_iter())
                     .ok()?;
-                Some(Proof {
-                    consensus_chain_block_info: BlockInfo {
-                        block_number: consensus_chain_block_info.block_number.into(),
-                        block_hash: consensus_chain_block_info.block_hash.into(),
-                    },
-                    consensus_chain_state_root: consensus_chain_state_root.into(),
-                    domain_proof: Some((
-                        BlockInfo {
-                            block_number: number,
-                            block_hash: hash,
+                // TODO: Derive correct domain proof
```

Comment:
Is there tracking for this TODO? please put a link if so. if not - well there should be :)

Reply:
This is todo is already resolved as part of the next PR
more here - https://github.com/subspace/subspace/pull/2562/files#diff-f6f9310595b3fedfb829530fe9971b90398313a38566a75f08479f8518db178fL384